src/main/docs/serializers.adoc

```diff
@@ -0,0 +1,265 @@
+ifndef::sourcedir[]
+:sourcedir: ../..
+endif::sourcedir[]
+
+= Custom Serializers
+
+The Ehcache 3 documentation at http://www.ehcache.org/documentation/3.0/serializers-copiers.html#serializers[Serializers]
+gives you an overview of how to use custom serializers with a cache.
+The section on http://www.ehcache.org/documentation/3.0/serializers-copiers.html#persistent-vs-transient-caches[Persistent and Transient Serializers]
+briefly covers the serializer contracts that must be honored while writing custom serializers to be used with 
+persistent/transient caches.
+
+This document explains how you can write a transient/persistent custom serilaizer that works with Ehcache.
+Here we discuss the significance of transient serializers and persistent serializers in detail and what they actually 
+mean.
+
+== Serializer types
+
+In the Ehcache getting started document you must have already seen the serializers requiring a single argument constructor
+or a double-argument constructor or both based on the type of the cache they are used in.
+The doc further explains that the serializer with the single-argument constructor is fit to be used with transient 
+caches and the ones with the double-argument constructor can be used with persistent caches. An implementation having
+both the constructors can be used with both _persistent_ and _transient_ caches.
+Hmm... So what does that really mean?
+If you look at the custom serializer implementations in the GettingStarted samples they are all have both the costructors
+and if you look at the code they don't do anything different.
+It's all standard java serialization.
+So what difference does the constructors make?
```

Comment:
does -> do
And again, most likely blank lines before and after to split paragraphs.